doc/classes/Engine.xml

```diff
@@ -4,16 +4,16 @@
 		Provides access to engine properties.
 	</brief_description>
 	<description>
-		The [Engine] singleton allows you to query and modify the project's run-time parameters, such as frames per second, time scale, and others.
+		The [Engine] singleton allows you to query and modify the project's run-time parameters, such as frames per second, time scale, and others. It also stores information about the current build of the Godot Engine, such as the current version.
 	</description>
 	<tutorials>
 	</tutorials>
 	<methods>
 		<method name="get_architecture_name" qualifiers="const">
 			<return type="String" />
 			<description>
-				Returns the name of the CPU architecture the Godot binary was built for. Possible return values are [code]x86_64[/code], [code]x86_32[/code], [code]arm64[/code], [code]arm32[/code], [code]rv64[/code], [code]riscv[/code], [code]ppc64[/code], [code]ppc[/code], [code]wasm64[/code] and [code]wasm32[/code].
-				To detect whether the current CPU architecture is 64-bit, you can use the fact that all 64-bit architecture names have [code]64[/code] in their name:
+				Returns the name of the CPU architecture the Godot binary was built for. Possible return values include [code]"x86_64"[/code], [code]"x86_32"[/code], [code]"arm64"[/code], [code]"arm32"[/code], [code]"rv64"[/code], [code]"riscv"[/code], [code]"ppc64"[/code], [code]"ppc"[/code], [code]"wasm64"[/code], and [code]"wasm32"[/code].
```

Comment:
Could it return `riscv64`?

Reply:
Does not seem like it.
https://github.com/godotengine/godot/blob/bb6b06c81343073f10cbbd2af515cf0dac1e6549/core/config/engine.cpp#L201-L235

Reply:
Sorry, it's `rv64`. \*@adamscott returned to his home planet.\*

Reply:
It's strange that it's using the different names from the `arch` values (do not have options for 32-bit RISC-V and 64-bit WASM). Probably should be changed to match the general pattern:

```diff
diff --git a/core/config/engine.cpp b/core/config/engine.cpp
index 2bb8837849..90e3d58a7e 100644
--- a/core/config/engine.cpp
+++ b/core/config/engine.cpp
@@ -215,14 +215,14 @@ String Engine::get_architecture_name() const {
 #if __riscv_xlen == 8
 	return "rv64";
 #else
-	return "riscv";
+	return "rv32";
 #endif
 
 #elif defined(__powerpc__)
 #if defined(__powerpc64__)
 	return "ppc64";
 #else
-	return "ppc";
+	return "ppc32";
 #endif
 
 #elif defined(__wasm__)
```

Reply:
Keep this patch in mind for another PR, hmm.